pkg/rtc/participant.go

```diff
@@ -1651,10 +1675,17 @@ func (p *ParticipantImpl) handlePendingDataChannels() {
 		if err != nil {
 			p.params.Logger.Errorw("create migrated data channel failed", err, "label", ci.Label)
 		} else if dc != nil {
-			p.onDataChannel(dc)
+			created = created || p.onDataChannelLocked(dc)
```

Comment:
if created is true, this expression will be short circuit, and p.OnDataChannelLocked will not be execute.
we have two dc here(lossy and reliable), so p.onDataChannelLocked will be only execute once, and another dc's OnMessage will not be listened for datatrack

Reply:
Ah, missed that. Fixed it in this commit - https://github.com/livekit/livekit-server/pull/508/commits/ef1cdcf58eab0055ba4c17553c838624eb96f2cf. Migration works now. Thank you @cnderrauber